### DISCUSSION TITLE
Changing your GitHub username.  Only concerned about Visual Studio synchronization and maintaining my Copilot Subscription.

### DISCUSSION THREAD

**Discussion by Axium7** [community/community#138138] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello GitHub,

I was considering changing my username.  I have only been using GitHub this month basically and will continue with my Copilot Subscription.   I am only concerned about my Repositories in Visual Studio still working with GitHub and continuing my Copilot Subscription.   How difficult will it be to reconfigure Visual Studio to work with the new username?  Will this affect my Copilot Subscription?

Thank you for your feedback.

**Comment by davevad93** [comment]
Hi @Picard78 ,

I have a free Copilot subscription since last year thanks to the `Student Developer Pack`. A few months ago, I changed my username and there hasn't been any problem at all; I just had to log out of the extension in Visual Studio 2022 and VS Code and log back in. If you're paying for Copilot, I assume it's the same because the subscription is linked to your billing plan, so as long as you don't change it, everything should be fine.

Changing your username can have unintended side effects., as indicated in the [docs](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/changing-your-github-username), but none related to the Copilot subscription.

**Reply by Axium7** [reply]
davevad93,  thank you for your answer in regards to Subscription.   What about the changes to Visual Studio and connecting to my Repos through the IDE?   Will I have to make some changes there or will GitHub take care of that for me?

**Reply by davevad93** [reply]
Hey @Picard78 you're welcome, in VS Code the repos stay synchronized, regarding Visual Studio I can't confirm it because when I changed my username I didn't have my repos synchronized with that IDE, but I guess it's the same.

**Reply by Axium7** [reply]
That is good news.  Thank you davevad93

**Reply by davevad93** [reply]
You're welcome @Picard78 👍

**Comment by hurradieweltgehtunter** [comment]
Do you have any idea, if it is affected when you have a free CoPilot subscription? I'm using CoPilot for free because I'm a maintainer in a popular open-source project. Now I need to change my username. Any idea if this affects the free CoPilot subscription?

**Comment by Axium7** [comment, marked as answer]
Hurra,

I just changed my GitHub username.  I am still on a free Trial CoPilot Subscription and I encountered no issues.   The service still works for me.  I also encountered no issues in Visual Studio Local Repos.  Everything seems to be working smoothly so far.

The username change was instantaneous.  So far I am happy I made the change.  Thank you David for providing your experience.  It gave me the confidence to make the change myself.

**Reply by davevad93** [reply]
Hey @Axium7 I'm so glad, happy to help! 👍

@hurradieweltgehtunter you can safely change your username and continue with your free Copilot subscription. 

Have a great day both of you!

**Reply by Axium7** [reply]
Thanks Dave.  Have a good weekend.

**Reply by hurradieweltgehtunter** [reply]
thank you very much :)
